extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -468,9 +572,29 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 			console.log(error);
 		}
 
+		// Generate telemetry for stop data collection request
+		this.generateStopDataCollectionTelemetry().catch(e => console.error(e));
 		return true;
 	}
 
+	private async generateStopDataCollectionTelemetry(): Promise<void> {
+		try {
+			sendSqlMigrationActionEvent(
+				TelemetryViews.DataCollectionWizard,
+				TelemetryAction.StopDataCollection,
+				{
+					'sessionId': this._sessionId,
+					'TimeDataCollectionStopped': this._perfDataCollectionStopDate?.toString()
```

Comment:
nit: can we lowercase to `timeDataCollectionStopped` to keep the formatting consistent